### PR TITLE
Import tests from Google V8 (Block Scope Additions)

### PR DESCRIPTION
**Pull request by rwaldron**
These tests are derived from the following files within the Google V8 project:

    test/mjsunit/es6/regress/regress-2506.js
    test/mjsunit/es6/regress/regress-3426.js
    test/mjsunit/es6/regress/regress-3683.js

### CODE REVIEW

File: test/language/block-scope/semantics/const/for-of-in-fresh-binding-per-iteration.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+assert.sameValue(s, 6, "The value of `s` is `6`");
+assert.sameValue(f[0](), 1, "`f[0]()` returns `1`");
+assert.sameValue(f[1](), 2, "`f[1]()` returns `2`");
+assert.sameValue(f[2](), 3, "`f[2]()` returns `3`");
+
+
+
+let x = 1;
+s = 0;
+for (const x of [x, x+1, x+2]) {
```
Comment (anba) [inline test/language/block-scope/semantics/const/for-of-in-fresh-binding-per-iteration.js]: This line throws a ReferenceError at runtime b/c of TDZ for `x` in `[x, x+1, x+2]` (13.6.4.12 ForIn/OfHeadEvaluation, step 2).

File: test/language/block-scope/semantics/const/for-of-in-fresh-binding-per-iteration.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  s += x;
+}
+assert.sameValue(s, 6, "The value of `s` is `6`");
+
+s = 0;
+var q = 1;
+for (const q of [q, q+1, q+2]) {
```
Comment (anba) [inline test/language/block-scope/semantics/const/for-of-in-fresh-binding-per-iteration.js]: Same issue here.

File: test/language/block-scope/semantics/const/for-of-in-fresh-binding-per-iteration.js (hunk 1 above)
Comment (rwaldron) [inline test/language/block-scope/semantics/const/for-of-in-fresh-binding-per-iteration.js]: Indeed it does, but these tests look very explicit, so hopefully @dslomov-chromium or @rossberg can chime in?

Comment (arv) [inline test/language/block-scope/semantics/const/for-of-in-fresh-binding-per-iteration.js]: @rossberg-chromium

Comment (dslomov-chromium) [inline test/language/block-scope/semantics/const/for-of-in-fresh-binding-per-iteration.js]: This definitely looks like a bug in V8.

Comment (dslomov-chromium) [inline test/language/block-scope/semantics/const/for-of-in-fresh-binding-per-iteration.js]: This is now issue 4000 in V8: https://code.google.com/p/v8/issues/detail?id=4000

Comment (rwaldron) [inline test/language/block-scope/semantics/const/for-of-in-fresh-binding-per-iteration.js]: Thanks @dslomov-chromium 


@anba updates to follow

**Comment by bterlson** [comment]
Content looks good. Organization is strange. IMO:

* language/block-scope/continue --> language/statements/continue
* language/block-scope/semantics/const --> language/statements/const, language/statements/for, language/statements/for-of, or language/statements/for-in (collapse all tests under here)

**Comment by bterlson** [comment]
If you want to preserve the block-scope metadata using the features frontmatter attribute seems an acceptable way to do it (call it block-scope I guess).

**Comment by rwaldron** [comment]
- Rebased
- Recommended relocations completed

**Comment by rwaldron** [comment]
I think I'll do more re-organization on this one.

**Comment by bterlson** [comment]
Let me know once done :)

**Comment by rwaldron** [comment]
I think this is ok now, unless there is anything you want to change

**Pull request merged**